db/compaction/compaction_picker_test.cc

```diff
@@ -1342,13 +1342,142 @@ TEST_F(CompactionPickerTest, CompactionPriRoundRobin) {
             cf_name_, mutable_cf_options_, mutable_db_options_, vstorage_.get(),
             &log_buffer_));
     ASSERT_TRUE(compaction.get() != nullptr);
+    // Since the max bytes for level 2 is 120M, picking one file to compact
+    // makes the post-compaction level size less than 120M, there is exactly one
+    // file picked for round-robin compaction
     ASSERT_EQ(1U, compaction->num_input_files(0));
     ASSERT_EQ(selected_files[i], compaction->input(0, 0)->fd.GetNumber());
     // release the version storage
     DeleteVersionStorage();
   }
 }
 
+TEST_F(CompactionPickerTest, CompactionPriMultipleFilesRoundRobin1) {
+  ioptions_.compaction_pri = kRoundRobin;
+  mutable_cf_options_.max_compaction_bytes = 100000000000u;
+  mutable_cf_options_.max_bytes_for_level_base = 12000000;
+  mutable_cf_options_.max_bytes_for_level_multiplier = 10;
+  // start a brand new version in each test.
+  NewVersionStorage(6, kCompactionStyleLevel);
+  vstorage_->ResizeCompactCursors(6);
+  // Set the cursor (file picking should start with 7U)
+  vstorage_->AddCursorForOneLevel(2, InternalKey("199", 100, kTypeValue));
+  Add(2, 6U, "150", "199", 50000000U);
+  Add(2, 7U, "200", "249", 50000000U);
+  Add(2, 8U, "300", "600", 50000000U);
+  Add(2, 9U, "700", "800", 50000000U);
+  Add(2, 10U, "850", "950", 50000000U);
+
+  Add(3, 26U, "130", "165", 60000000U);
+  Add(3, 27U, "166", "170", 60000000U);
+  Add(3, 28U, "270", "340", 60000000U);
+  Add(3, 29U, "401", "500", 60000000U);
+  Add(3, 30U, "601", "800", 60000000U);
+  Add(3, 31U, "830", "890", 60000000U);
```

Comment:
Will it still work if we delete the trailing six zeroes from all these numbers? The numbers are hard to read.